apps/labs/posts/automated-accessibility-testing-with-bokeh.md

```diff
@@ -0,0 +1,178 @@
+
+## What is Accessibility?
+
+Accessibility in this context refers to the practice of designing tools in ways
+that people with disabilities would be able to perceive, understand, navigate,
+and interact with them.
+According to the [World Health Organization's study in
+2023](https://www.who.int/news-room/fact-sheets/detail/disability-and-health),
+roughly 16% of the world's population, or about 1.6 billion people experience
+significant disability. This essentially means we have to do better by way of
+building usable software for these people, and this starts with concerted
+efforts towards accessibility.
+
+## Proposed Work
+
+The work I did was part of a many-fold proposal. It ran asynchronously (and
+somewhat concurrently) with Bokeh's overall goal of making data visualizations
+more accessible.
+On one end, we had [Frank Elavsky](https://www.frank.computer/) conducting
+accessibility research with help from the Bokeh maintainers and Quansight team.
+On the other end, I was trying to set up a suitable test environment in the
+codebase. The idea was the research was completed and an audit was performed,
+and the teams from Bokeh and Quansight (as well as any interested contributor)
+could start making fixes to the pertinent parts of Bokeh. For every fix, there
+would need to be an accompanying test to avoid regression later on.
+The problem, though, was that much of Bokeh runs on custom-built solutions. It
+had its own setup for almost everything. It required specific types of
+environments to match its use cases, it had a custom compilation, build & run
+tooling, and they weren't as thoroughly documented. Bokeh itself has good
+documentation for users, but the internal documentation of how things worked
+wasn't really there. What this meant for my project was that it would be more of
+research work, trying to come up with something to fit into what Bokeh had, with
+the least amount of friction. This, while not sacrificing functionality.
+I'll have to mention that prior to this project, there was no provision for
+Accessibility testing in the Bokeh codebase, and the closest we had to that was
+removed sometime in the past because it had been unused and unmaintained for so
+long. The primary goal of my work would thus be to move Bokeh from 0 to 1 and
+give feedback and findings that could be built upon. Essentially, the spark that
+starts the fire.
+
+## How Bokeh works under the hood
+
+Talking about the scope of what I worked on and the accompanying complexities
+wouldn't make as much sense if I didn't explain the mechanism behind Bokeh. To
+the typical user who needs to get work done, you just install the package, type
+some commands as instructed by the docs, and then run. In doing this, if you
+look closely at your file directly you'll realize that a matching HTML file has
+been created and automatically opens in your browser. Voila! You get an
+interactive plot. With a few lines of Python code, you've done some apparent
+magic.
```

Comment:
Let's try to avoid making anything sound simple. It's very frustrating in docs and in technical writing for the author to say that something is easy if you then end up struggling with it.